### ISSUE TITLE
feat:  use mimetype to determine filetype in linux

### ISSUE THREAD

**Issue by Rid1FZ** [eza-community/eza#1073]
As far as I noticed, eza uses some sort of regex or file extension to determine filetype. But it is very common in linux to fully avoid extension for any script for example. In that cases, eza shows default icon for that file (which is incorrect). Is it possible to use mimetype instead for filetype detectiom in linux? Mimetype of any file is detected by reading the file instead of using anything like regex. It helps to determine filetype more precisely.

**Comment by gierens** [comment]
You are correct, at the moment the filename/extension is used to determine the type. We are not using the mime/content type as far as I'm aware.

As for adding this functionality, sure that is possible, we would first have to check how to do it efficiently, though. If we really have to read the file, even just partly, this will likely incur a decent amount of overhead, so it would probably be best to put it behind a command line flag to enable it rather than making it the default.

**Comment by cafkafk** [comment]
> As for adding this functionality, sure that is possible, we would first have to check how to do it efficiently, though. If we really have to read the file, even just partly, this will likely incur a decent amount of overhead, so it would probably be best to put it behind a command line flag to enable it rather than making it the default behavior.

Ideally we would also start measuring that overhead, so we could make informed choices... but then again, we really need some way to catch performance regressions in reviews
